pgt-leader-bot/templates/deployment.yaml

```diff
@@ -0,0 +1,96 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "pgt-leader-bot.fullname" . }}-api
+  {{- with .Values.deploymentAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    {{- include "pgt-leader-bot.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "pgt-leader-bot.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        component: api
```

Comment:
Awful hack to fix the issue for the bot to contact the api via the service.

We need to implement a better solution. Also, GPT Survey Summarizer need to be revisited, as a suffix is added on a label selector map, which is inconsistent.